adds switch for base models based on etl

ChrisLawliss opened the pull request:
# Fivetran Compatibility
- create fivetran macros
- update models to pull from macros specific to the ETL specified at the project level

ChrisLawliss committed:
adds switch for base models based on etl
adds fivetran macros for base models

ChrisLawliss committed:
deleted models based on things fivetran does not give, removed redundant aliasing

ChrisLawliss committed:
updated vars for etiher sttich or fivetran

ChrisLawliss committed:
removed references to core table macros for fivetran

ChrisLawliss committed:
moved adapters and models directory up a level

eogilvy12 commented on models/stitch/base/adwords_ad_groups.sql:
```diff
@@ -1 +1,5 @@
+{% if var('etl') == 'stitch' %}
+{{ stitch_adwords_ad_groups() }}
+{% elif var('etl') == 'fivetran'%}
+--Fivetran does not support Core tables
```
same as other comment

eogilvy12 commented on models/stitch/base/adwords_accounts.sql:
```diff
@@ -1 +1,5 @@
+{% if var('etl') == 'stitch' %}
+{{ stitch_adwords_accounts() }}
+{% elif var('etl') == 'fivetran'%}
+--Fivetran does not support Core tables
```
instead of even listing the fivetran option, I think the default for fivetran should be to disable the core tables in the`dbt_project.yml` file. you can comment these options out since the default will be stitch

eogilvy12 commented on models/stitch/base/adwords_ads.sql:
```diff
@@ -1 +1,5 @@
+{% if var('etl') == 'stitch' %}
+{{ stitch_adwords_ads() }}
+{% elif var('etl') == 'fivetran'%}
+--Fivetran does not support Core tables
```
same as other comment

eogilvy12 commented on models/stitch/base/adwords_click_performance.sql:
```diff
@@ -1 +1,5 @@
+{% if var('etl') == 'stitch' %}
+{{ stitch_adwords_click_performance() }}
+{% elif var('etl') == 'fivetran'%}
+{{ fivetran_adwords_click_performance() }}
```
same as other comment

eogilvy12 reviewed (changes requested):
some small changes around the core tables and disabling in the project file

ChrisLawliss committed:
adds commented out disabling of core tables in dbt_project.yml for easy disabling
removes references to fivetran in core models
deletes extra new lines at the end of macros

ChrisLawliss committed:
removed  logic from fivetran adwords macros as conversion happens automatically

eogilvy12 reviewed (approved):
this looks good to me - @jthandy can you take a final look over this? (again would love a second set of eye prior to this large incorporation for fivetran!)

ChrisLawliss committed:
updated base models to use router structure

jthandy commented on models/adapters/adwords_ad_performance_adapter.sql:
```diff
@@ -1,8 +1,11 @@
 {% if var('adapter_value') == 'url' %} 
```
this model seems to still only be stitch-specific, right? from what I can tell, there is only one version of this model (the stitch one).

This is actually a "transform" model--I don't think it touches the raw data. So I think the right answer is to implement it in your "router" pattern. I'm not 100% sure here, but happy to discuss with you.

ChrisLawliss replied:
@jthandy Check out the latest commit. I moved the logic to the routers, _however_ `fivetran/adwords_criteria_performance` is now identical to `fivetran/adapters/criteria/adwords_performance_adapter`. Let me know what you think the right step forward is.

jthandy reviewed (changes requested):
This looks super super solid. I have one comment, and I'm happy to discuss approaches with you in person.

ChrisLawliss committed:
moved adapter logic to routers

ChrisLawliss committed:
updated ad_performance adapter to be unioned with bing and facebook ads packages

ChrisLawliss committed:
formatting

ChrisLawliss commented:
updated adapter models to use the router pattern. Also renamed fields to be unioned with other ad packages. @eogilvy12 @jthandy These changes are ready for review

ChrisLawliss committed:
reorganized router for simplicity
